vllm/transformers_utils/config.py

```diff
@@ -740,6 +757,11 @@ def get_hf_text_config(config: PretrainedConfig):
         # if transformers config doesn't align with this assumption.
         assert hasattr(config.text_config, "num_attention_heads")
         return config.text_config
+    elif hasattr(config, "thinker_config"):
+        # TODO(suyang.fy): Refactor code.
+        #  For Qwen2.5-Omni, change hf_text_config to
+        #  thinker_config.text_config.
+        return config.thinker_config.text_config
```

Comment:
Based on discussion in https://github.com/huggingface/transformers/issues/37515#issuecomment-2805020779 it might be better to use the `get_text_config()` in future so that the attr to use is determined on the HF side.

This util doesn't work for Qwen2.5-Omni now, but it should in a future release.

Reply:
For posterity, I have made this change in https://github.com/vllm-project/vllm/pull/21057